monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,618 @@
+from monai.utils.misc import label_union
+
+yaml, _ = optional_import("yaml")
+tqdm, has_tqdm = optional_import("tqdm", "4.47.0", min_version, "tqdm")
+measure, _ = optional_import("scipy.ndimage.measurements")
+
+logger = get_logger(module_name=__name__)
+
+__all__ = ["DataAnalyzer"]
+
+
+class DataAnalyzer:
+    """
+    The DataAnalyzer automatically analyzes given medical image dataset and reports the statistics.
+    The module expects file paths to the image data and utilizes the LoadImaged transform to read the files.
+    which supports nii, nii.gz, png, jpg, bmp, npz, npy, and dcm formats. Currently, only segmentation
+    problem is supported, so the user needs to provide paths to the image and label files. Also, label
+    data format is preferred to be (1,H,W,D), with the label index in the first dimension. If it is in
+    onehot format, it will be converted to the preferred format.
+
+    Args:
+        datalist: a Python dictionary storing group, fold, and other information of the medical
+            image dataset, or a string to the JSON file storing the dictionary.
+        dataroot: user's local directory containing the datasets.
+        output_path: path to save the analysis result.
+        average: whether to average the statistical value across different image modalities.
+        do_ccp: apply the connected component algorithm to process the labels/images
+        device: a string specifying hardware (CUDA/CPU) utilized for the operations.
+        worker: number of workers to use for parallel processing.
+
+    For example:
+
+    .. code-block:: python
+
+        from monai.apps.auto3d.data_analyzer import DataAnalyzer
+
+        datalist = {
+            "testing": [{"image": "image_003.nii.gz"}],
+            "training": [
+                {"fold": 0, "image": "image_001.nii.gz", "label": "label_001.nii.gz"},
+                {"fold": 0, "image": "image_002.nii.gz", "label": "label_002.nii.gz"},
+                {"fold": 1, "image": "image_001.nii.gz", "label": "label_001.nii.gz"},
+                {"fold": 1, "image": "image_004.nii.gz", "label": "label_004.nii.gz"},
+            ],
+        }
+
+        dataroot = '/datasets' # the directory where you have the image files (nii.gz)
+        DataAnalyzer(datalist, dataroot)
+
+    Note:
+        The module can also be called from the command line interface (CLI) if MONAI is
+        installed with the source code.
+
+    For example:
+
+    .. code-block:: bash
+
+        python monai/apps/auto3d \
+            DataAnalyzer \
+            get_all_case_stats \
+            --datalist="my_datalist.json" \
+            --dataroot="my_dataroot_dir"
+
+    """
+
+    def __init__(
+        self,
+        datalist: Union[str, Dict],
+        dataroot: str,
+        output_path: str = "./data_stats.yaml",
+        average: bool = False,
+        do_ccp: bool = True,
+        device: Union[str, torch.device] = "cuda",
+        worker: int = 2,
```

Comment:
I try the example code in python, but get this error with mac os:
```
    for batch_data in tqdm(self.dataset) if has_tqdm else self.dataset:
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/site-packages/tqdm/std.py", line 1195, in __iter__
    for obj in iterable:
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 444, in __iter__
    return self._get_iterator()
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 390, in _get_iterator
    return _MultiProcessingDataLoaderIter(self)
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 1077, in __init__
    w.start()
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
AttributeError: Can't pickle local object 'DataAnalyzer.__init__.<locals>.<lambda>'
```
then I set `worker=0`, it gives another one:
```
  File "/usr/local/anaconda3/envs/py38/lib/python3.8/site-packages/torch/cuda/__init__.py", line 211, in _lazy_init
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled
```